src/Adapter/Order/OrderDetailUpdater.php

```diff
@@ -0,0 +1,471 @@
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Order;
+
+use Address;
+use Cart;
+use Country;
+use Currency;
+use Customer;
+use Db;
+use Language;
+use Order;
+use OrderDetail;
+use PrestaShop\Decimal\Number;
+use PrestaShop\PrestaShop\Adapter\ContextStateManager;
+use PrestaShop\PrestaShop\Core\Domain\Configuration\ShopConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use PrestaShop\PrestaShop\Core\Localization\CLDR\ComputingPrecision;
+use Product;
+use Shop;
+use TaxCalculator;
+use TaxManagerFactory;
+use Tools;
+
+class OrderDetailUpdater
+{
+    private const COMPARISON_PRECISION = 6;
+
+    /**
+     * @var ContextStateManager
+     */
+    private $contextStateManager;
+
+    /**
+     * @var ShopConfigurationInterface
+     */
+    private $shopConfiguration;
+
+    /**
+     * @param ContextStateManager $contextStateManager
+     * @param ShopConfigurationInterface $shopConfiguration
+     */
+    public function __construct(
+        ContextStateManager $contextStateManager,
+        ShopConfigurationInterface $shopConfiguration
+    ) {
+        $this->contextStateManager = $contextStateManager;
+        $this->shopConfiguration = $shopConfiguration;
+    }
+
+    /**
+     * @param OrderDetail $orderDetail
+     * @param Order $order
+     * @param Number $priceTaxExcluded
+     * @param Number $priceTaxIncluded
+     *
+     * @throws OrderException
+     */
+    public function updateOrderDetail(
+        OrderDetail $orderDetail,
+        Order $order,
+        Number $priceTaxExcluded,
+        Number $priceTaxIncluded
+    ): void {
+        list($roundType, $computingPrecision, $taxAddress) = $this->prepareOrderContext($order);
+
+        try {
+            $precisePriceTaxExcluded = $this->getPrecisePriceTaxExcluded($priceTaxIncluded, $priceTaxExcluded, $order, $orderDetail, $taxAddress);
+            $precisePriceTaxIncluded = $this->getPrecisePriceTaxIncluded($priceTaxIncluded, $priceTaxExcluded, $order, $orderDetail, $taxAddress);
+
+            $this->applyPriceUpdate(
+                $orderDetail,
+                $precisePriceTaxExcluded,
+                $precisePriceTaxIncluded,
+                $roundType,
+                $computingPrecision
+            );
+        } finally {
+            $this->contextStateManager->restorePreviousContext();
+        }
+    }
+
+    /**
+     * @param Order $order
+     * @param int $productId
+     * @param int $combinationId
+     * @param Number $priceTaxExcluded
+     * @param Number $priceTaxIncluded
+     *
+     * @throws OrderException
+     */
+    public function updateIdenticalOrderDetails(
```

Comment:
FMI, I'm not sure I understand the concept of "identical order details", is this a tricky/specific case, or am I missing something? 🤔

Reply:
Ok, I answer myself :D, from what I see in the code it's the other order details having the same product/combination.

Maybe a short comment explaining what is an "identical order detail" would help 👍

Reply:
yeah I guess naming can be improved why not `updateOrderDetailsForProduct` ?

Reply:
ah yes, that sounds good, or even `updateOrderDetailsForSameProduct`, so we quickly understand that it's order details having the same product

Reply:
Well actually `updateOrderDetailsForProduct` is better, my suggestion is confusing and could be misunderstood 👍